presentation/common/src/presentation-common/Update.ts

```diff
@@ -118,67 +119,97 @@ export type PartialHierarchyModification = NodeInsertionInfo | NodeDeletionInfo
 export namespace PartialHierarchyModification { // eslint-disable-line @typescript-eslint/no-redeclare
   /** Serialize given object to JSON. */
   export function toJSON(obj: PartialHierarchyModification): PartialHierarchyModificationJSON {
-    return {
-      ...obj,
-      node: Node.toJSON(obj.node),
-    };
+    switch (obj.type) {
+      case "Insert":
+        return {
+          type: "Insert",
+          node: Node.toJSON(obj.node),
+          position: obj.position,
+          parent: obj.parent,
+        };
+
+      case "Update":
+        return {
+          type: "Update",
+          target: NodeKey.toJSON(obj.target),
+          changes: Node.toPartialJson(obj.changes),
+        };
+
+      case "Delete":
+        return {
+          type: "Delete",
+          target: NodeKey.toJSON(obj.target),
+        };
+    }
   }
 
   /** Deserialize given object from JSON */
   export function fromJSON(json: PartialHierarchyModificationJSON): PartialHierarchyModification {
-    return {
-      ...json,
-      node: Node.fromJSON(json.node),
-    };
+    switch (json.type) {
+      case "Insert":
+        return {
+          type: "Insert",
+          node: Node.fromJSON(json.node),
+          position: json.position,
+          parent: json.parent,
+        };
+
+      case "Update":
+        return {
+          type: "Update",
+          target: NodeKey.fromJSON(json.target),
+          changes: Node.fromPartialJson(json.changes),
+        };
+
+      case "Delete":
+        return {
+          type: "Delete",
+          target: NodeKey.fromJSON(json.target),
+        };
+    }
   }
 }
 
 /** @alpha */
 export interface NodeInsertionInfo {
   type: "Insert";
+  parent?: string;

```

Comment:
I think this should be `parent?: NodeKey`

Reply:
Thanks, fixed in 8ae2f2b242a47058e52e6794e9850e95aeecec6d.